Aria-label for UI Tree

tfamula opened the pull request:
This is a follow-up of one open task, which was discussed in https://github.com/ILIAS-eLearning/ILIAS/pull/2500. 
It tackles the UI Tree, which should get a new (aria-)label to improve accessibility. Because the UI Tree is already used in a few places, this PR includes also suggestions for existing Trees, where the labels have to be added retroactively.

tfamula committed:
Aria-label for UI Tree

mjansenDatabay reviewed (approved):
Thx @tfamula , the changes in my consuming components LGTM!

IMO the `label` should be the 2nd parameter when retrieving the tree from the factory, but tha's just my personal taste (maybe because it is a primitive type).

alex40724 reviewed (approved):
+1 Thanks @tfamula !

Amstutz commented:
Hi @tfamula 

Thanks a lot for this PR. As far as I understand the linked guideline in the PR above, I believe this to be a step in the right direction of making ILIAS more accessible. Kudos!

This is just a review on the interface, as soon as accepted by the JF, I will also provide one on the implementation:

Improve or argue why not:
- [x] Some more detail in Rule: The proposed rule:    
`*The "aria-label" attribute MUST be set for Expandable Trees, which MUST be language-dependant.`

Is good, but to me it should also provide help in finding a good label. We should make clear here, the simply providing e.g. "Tree" is not good enough. I propose to rephrase to:
```
*The "aria-label" attribute MUST be set for Expandable Trees
*The "aria-label" MUST be language-dependant.
*The "aria-label" MUST describe the content of the tree as precisely as possible. "Tree" is not a good label, "Forum Posts" is much more helpful. Note that the "Tree" is already set by the aria-role attribute.
```

Since the interface change is not large and therefore not too complex to discuss, I put this on the JF today. This is on very short notice, if anybody needs more time to review the interface change here, feel free to ask for it.

klees commented:
Hi @tfamula.

what @Amstutz says + Please keep the order of the parameters of the constructor (other then what @mjansenDatabay suggests). There are many components where the label comes first in the parameter list, I think this is a good convention to maintain.

Thanks!

matthiaskunkel commented:
Jour Fixe, 17 AUG 2020: We highly appreciate this PR and accept it for trunk and ILIAS 6 (bug fix). Please merge.

tfamula committed:
Update UI Tree rules

tfamula commented:
Hi @Amstutz and @klees 
Thanks for your feedback! I agree with everything and updated the rules for the Tree according to Timon's suggestion. Please have a quick look, because I adjusted the wording a little bit.

Amstutz commented:
Hi @tfamula 

Nice, thx a lot. looking good. Is the complete implementation ready for the final review?

Thx.

tfamula commented:
@Amstutz Yes, it is. The tests are added / modified, too.

Amstutz commented:
Thx a lot, I just cherry-picked this down to 6 as well.
